### DISCUSSION TITLE
status of FHIR StructureDefinitions

### DISCUSSION THREAD

**Discussion by bradlittleton** [b2ihealthcare/snow-owl#933] [Q&A] [answered]
Hi,

I was wondering what the status of the Structure Definition resource was?  I can see from the swagger docs that there are some endpoints in the REST API to retrieve structure definitions, but no way to create them.  Is there another way to load them, or is this planned for the future?

Cheers,
Brad

**Comment by cmark** [comment, marked as answer]
Hi @bradlittleton,

Thank you for your question.
That FHIR resource is in a very early stage of development, only the read API endpoints are there, no actual CUD operations are available yet. There is no option to load them in any other way at the moment, this is purely because there were no requirements for it yet and we were just experimenting with the resource model itself. Do you have a specific use case or a few examples that could help us develop this feature further after the upcoming 8.0 major release?

Cheers,
Mark

**Reply by bradlittleton** [reply]
Hi @cmark ,

Thanks for your answer.  Our primary use case it that we exchange data with external health care providers via FHIR, and would like to validate the incoming FHIR bundles against our own structure definitions to catch any we can't ingest properly, or that aren't using the agreed codesystems, before they move further through the system.  We've looked at a few ways of doing it and, as we are using snowowl for local codesystem authoring, it would be convenient if we could also use it for FHIR validation.  It would also need to be serving LOINC and the various HL7 codesystems, but I believe that's possible?  It's good to hear that more is planned for the FHIR service down the track.

Cheers,
Brad

**Reply by cmark** [reply]
Hi @bradlittleton,

Nice to hear that use case for StructureDefinition. Thank you for sharing.

> as we are using snowowl for local codesystem authoring
Just wondering, is this Snow Owl our official GA client with version 4.6.x?

> It would also need to be serving LOINC and the various HL7 codesystems, but I believe that's possible?
Snow Owl does support LOINC and custom Local Code Systems (and expose them as FHIR Code System resources), but those are only available in our paid versions. Feel free to contact us at info AT b2i DOT sg.

Cheers,
Mark
